crates/syntax/src/ptr.rs

```diff
@@ -21,6 +21,10 @@ impl SyntaxNodePtr {
         SyntaxNodePtr { range: node.text_range(), kind: node.kind() }
     }
 
+    pub fn text_range(&self) -> TextRange {
+        self.range.clone()
+    }
```

Comment:
this should be kept private, we want to be able to experiemtn with other ptr representations, for exapmle, "node's number in dfs order".